compiler/qsc_circuit/src/circuit.rs

```diff
@@ -0,0 +1,612 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use rustc_hash::FxHashMap;
+use serde::Serialize;
+use std::{fmt::Display, fmt::Write, ops::Not, vec};
+
+/// Representation of a quantum circuit.
+/// Implementation of <https://github.com/microsoft/quantum-viz.js/wiki/API-schema-reference>
+#[derive(Clone, Serialize, Default, Debug, PartialEq)]
+pub struct Circuit {
+    pub operations: Vec<Operation>,
+    pub qubits: Vec<Qubit>,
+}
+
+#[derive(Clone, Serialize, Debug, PartialEq)]
+pub struct Operation {
+    #[allow(clippy::struct_field_names)]
+    pub gate: String,
+    #[serde(rename = "displayArgs")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub display_args: Option<String>,
+    #[serde(rename = "isControlled")]
+    #[serde(skip_serializing_if = "Not::not")]
+    pub is_controlled: bool,
+    #[serde(rename = "isAdjoint")]
+    #[serde(skip_serializing_if = "Not::not")]
+    pub is_adjoint: bool,
+    #[serde(rename = "isMeasurement")]
+    #[serde(skip_serializing_if = "Not::not")]
+    pub is_measurement: bool,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub controls: Vec<Register>,
+    pub targets: Vec<Register>,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub children: Vec<Operation>,
+}
+
+static QUANTUM_REGISTER: usize = 0;
+static CLASSICAL_REGISTER: usize = 1;
```

Comment:
These two can be const.
```suggestion
const QUANTUM_REGISTER: usize = 0;
const CLASSICAL_REGISTER: usize = 1;
```